From 822add5d6653e32e8f4f25445a12c395c57e48ab Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Fri, 26 Feb 2021 19:09:52 +0530
Subject: [PATCH 1278/1921] octeontx-af: Add debug info on Mac address change

AF driver rejects change request of mac address from vf
if its set by admin(PF). Add debug info for the same.

Change-Id: I816256c56a9fffd81aad9d951af64c9fdbe4e3e8
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46831
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 790d67a1c988..f8d088f52d63 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -3461,8 +3461,11 @@ int rvu_mbox_handler_nix_set_mac_addr(struct rvu *rvu,
 	pfvf = rvu_get_pfvf(rvu, pcifunc);
 
 	/* VF can't overwrite admin(PF) changes */
-	if (from_vf && pfvf->pf_set_vf_cfg)
+	if (from_vf && pfvf->pf_set_vf_cfg) {
+		dev_warn(rvu->dev,
+			 "MAC address set by admin(PF) cannot be overwritten by VF");
 		return -EPERM;
+	}
 
 	ether_addr_copy(pfvf->mac_addr, req->mac_addr);
 
-- 
2.31.1

